src/index.ts

```diff
@@ -61,33 +62,35 @@ export interface AutoRetryOptions {
  * @returns The created API transformer function
  */
 export function autoRetry(
-    options?: Partial<AutoRetryOptions>
+  options?: Partial<AutoRetryOptions>,
 ): AutoRetryTransformer {
-    const maxDelay = options?.maxDelaySeconds ?? 3600
-    const maxRetries = options?.maxRetryAttempts ?? 3
-    const retryOnInternalServerErrors =
-        options?.retryOnInternalServerErrors ?? false
-    return async (prev, method, payload, signal) => {
-        let remainingAttempts = maxRetries
-        let result = await prev(method, payload, signal)
-        while (!result.ok && remainingAttempts-- > 0) {
-            let retry = false
-            if (
-                typeof result.parameters?.retry_after === 'number' &&
-                result.parameters.retry_after <= maxDelay
-            ) {
-                debug(`Hit rate limit, will retry '${method}' after ${result.parameters.retry_after} seconds`);
-                await pause(result.parameters.retry_after)
-                retry = true
-            } else if (
-                result.error_code >= 500 &&
-                retryOnInternalServerErrors
-            ) {
-                retry = true
-            }
-            if (!retry) return result
-            else result = await prev(method, payload, signal)
-        }
-        return result
+  const maxDelay = options?.maxDelaySeconds ?? Infinity;
+  const maxRetries = options?.maxRetryAttempts ?? Infinity;
+  const rethrowInternalServerErrors = options?.rethrowInternalServerErrors ??
+    false;
+  return async (prev, method, payload, signal) => {
+    let remainingAttempts = maxRetries;
+    let result = await prev(method, payload, signal);
+    while (!result.ok && remainingAttempts-- > 0) {
+      let retry = false;
+      if (
+        typeof result.parameters?.retry_after === "number" &&
+        result.parameters.retry_after <= maxDelay
+      ) {
+        debug(
+          `Hit rate limit, will retry '${method}' after ${result.parameters.retry_after} seconds`,
+        );
+        await pause(result.parameters.retry_after);
+        retry = true;
+      } else if (
+        result.error_code >= 500 &&
+        !rethrowInternalServerErrors
+      ) {
+        retry = true;
```

Comment:
Should we make any sane small pause, presuming servers may stay down on 5xx errors for longer than immediately, instead of retrying urgently?

Reply:
damn, yes